core/block.js

```diff
@@ -157,8 +158,13 @@ Blockly.Block = function(workspace, prototypeName, opt_id) {
   }
   // Bind an onchange function, if it exists.
   if (goog.isFunction(this.onchange)) {
-    this.onchangeWrapper_ = this.onchange.bind(this);
-    this.workspace.addChangeListener(this.onchangeWrapper_);
+    this.setOnChange(this.onchange);
+  }
+  // Apply extensions, if any.
+  if (Array.isArray(this.extensions)) {
```

Comment:
Turns out this wasn't doing anything.  The goog.mixin above is for the prototype definition object, not the JSON (which is wrapped in an init function, at this point).
Removed.